### ISSUE TITLE
Prune dependencies in starters

### ISSUE THREAD

**Issue by dsyer** [spring-cloud/spring-cloud-starters#10]
We tried pruning the dependencies in a minimal (empty) Spring Cloud app once and got the resulting ueberjar down to just over 20MB IIRC. We should do that again and bake the findings into starters.

**Comment by spencergibb** [comment]
Here is the dependency section of that app recorded for posterity
```
    <dependencies>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-starter-eureka</artifactId>
            <exclusions>
                <exclusion>
                    <groupId>com.google.inject.extensions</groupId>
                    <artifactId>guice-grapher</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>com.google.inject.extensions</groupId>
                    <artifactId>guice-assistedinject</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.codehaus.jsr166-mirror</groupId>
                    <artifactId>jsr166y</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>com.google.inject</groupId>
                    <artifactId>guice</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.apache.xbean</groupId>
                    <artifactId>xbean-finder</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>asm</groupId>
                    <artifactId>asm-all</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>com.google.inject.extensions</groupId>
                    <artifactId>guice-multibindings</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.apache.xbean</groupId>
                    <artifactId>xbean-bundleutils</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>javax.inject</groupId>
                    <artifactId>javax.inject</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.apache.commons</groupId>
                    <artifactId>commons-math</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>com.netflix.netflix-commons</groupId>
                    <artifactId>netflix-infix</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>joda-time</groupId>
                    <artifactId>joda-time</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>commons-jxpath</groupId>
                    <artifactId>commons-jxpath</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.antlr</groupId>
                    <artifactId>antlr-runtime</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.antlr</groupId>
                    <artifactId>stringtemplate</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>antlr</groupId>
                    <artifactId>antlr</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>com.google.code.gson</groupId>
                    <artifactId>gson</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>com.google.code.findbugs</groupId>
                    <artifactId>annotations</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>com.squareup.dagger</groupId>
                    <artifactId>dagger</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>com.google.inject</groupId>
            <artifactId>guice</artifactId>
            <version>2.0</version>
        </dependency>
    </dependencies>
```

I'm working on getting it into starters.

**Comment by spencergibb** [comment]
```
% ll -h target/testdeps-1.0-SNAPSHOT.jar
-rw-r--r-- 1 sgibb staff 25M Jan  9 11:00 target/testdeps-1.0-SNAPSHOT.jar
```

**Comment by spencergibb** [comment]
with just `spring-cloud-starter-hystrix`
``` 
ls -lh target/testdeps-1.0-SNAPSHOT.jar
-rw-r--r-- 1 sgibb staff 21M Jan  9 15:23 target/testdeps-1.0-SNAPSHOT.jar
```

**Comment by spencergibb** [comment]
with just `spring-cloud-starter-ribbon`
```
$ ls -lh target/testdeps-1.0-SNAPSHOT.jar
-rw-r--r-- 1 sgibb staff 23M Jan  9 15:27 target/testdeps-1.0-SNAPSHOT.jar
```

**Comment by spencergibb** [comment]
with just `spring-cloud-starter-feign`
```
% ls -lh target/testdeps-1.0-SNAPSHOT.jar
-rw-r--r-- 1 sgibb staff 21M Jan  9 15:29 target/testdeps-1.0-SNAPSHOT.jar
```

**Comment by spencergibb** [comment]
with just `spring-cloud-starter-turbine`
```
% ll -h target/testdeps-1.0-SNAPSHOT.jar                                (8074)[15:31:32]
-rw-r--r-- 1 sgibb staff 27M Jan  9 15:31 target/testdeps-1.0-SNAPSHOT.jar
```
brings in eureka-client.

**Comment by spencergibb** [comment]
with just `spring-cloud-starter-zuul`

```
$ ls -lh target/testdeps-1.0-SNAPSHOT.jar
-rw-r--r-- 1 sgibb staff 25M Jan  9 15:33 target/testdeps-1.0-SNAPSHOT.jar
```

**Comment by spencergibb** [comment]
with just `spring-cloud-starter-eureka-server`

```
% ls -lh target/testdeps-1.0-SNAPSHOT.jar
-rw-r--r-- 1 sgibb staff 39M Jan  9 16:31 target/testdeps-1.0-SNAPSHOT.jar
```

**Comment by spencergibb** [comment]
with just `spring-cloud-starter-hystrix-dashboard`

```
% ls -lh target/testdeps-1.0-SNAPSHOT.jar
-rw-r--r-- 1 sgibb staff 20M Jan  9 16:46 target/testdeps-1.0-SNAPSHOT.jar
```

**Comment by spencergibb** [comment]
fixed via 3e076411f0a30c35de4f8caaf60c6b8bfa7ecb59, 218eb59df80fa8edae6a954d81646ba012f84804 and 53c2666227846789e70e7e3076a2a5e24a33b6ce

Probably could do something with git to be one, oh well.
